utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp

```diff
@@ -901,33 +908,43 @@ int test(int test_case, int reader_type, const char *path, const char *outName,
         auto last_colot_temp = rocalGetIntValue(color_temp_adj);
         rocalUpdateIntParameter(last_colot_temp + 1, color_temp_adj);
 
-        rocalCopyToOutput(handle, mat_input.data, h * w * p);
+        RocalTensorList output_tensor_list = rocalGetOutputTensors(handle);
 
         std::vector<int> compression_params;
         compression_params.push_back(IMWRITE_PNG_COMPRESSION);
         compression_params.push_back(9);
 
-        mat_input.copyTo(mat_output(cv::Rect(col_counter * w, 0, w, h)));
-        std::string out_filename = std::string(outName) + ".png";   // in case the user specifies non png filename
-        if (display_all)
-            out_filename = std::string(outName) + std::to_string(index) + ".png";   // in case the user specifies non png filename
+        for(unsigned idx = 0; idx < output_tensor_list->size(); idx++) { // Iterate over each output
+            auto output_tensor = output_tensor_list->at(idx);
+            int h = output_tensor->shape().at(1) * output_tensor->dims().at(0);
+            int w = output_tensor->shape().at(0);
+            if(first_run) { // Allocate cv matrix and output buffers for each output in first run and reuse in every iteration
+                mat_output.emplace_back(cv::Mat(h, w, cv_color_format));
+            }            
+            output_tensor->copy_data(mat_output[idx].data, ROCAL_MEMCPY_HOST);  // Copy the output data into Opencv matrix
 
-        if (color_format == RocalImageColor::ROCAL_COLOR_RGB24)
-        {
-            cv::cvtColor(mat_output, mat_color, CV_RGB2BGR);
-            if(DISPLAY)
-                cv::imshow("output",mat_output);
-            else
-                cv::imwrite(out_filename, mat_color, compression_params);
-        }
-        else
-        {
-            if(DISPLAY)
-            cv::imshow("output",mat_output);
+            std::string out_filename = std::string(outName) + ".png";   // in case the user specifies non png filename
+            if (display_all)
+                out_filename = std::string(outName) + std::to_string(index) + std::to_string(idx) + ".png";   // in case the user specifies non png filename
+
+            if (color_format == RocalImageColor::ROCAL_COLOR_RGB24)
+            {
+                cv::cvtColor(mat_output[idx], mat_color, CV_RGB2BGR);
+                if(DISPLAY)
+                    cv::imshow("output", mat_output[idx]);
+                else
+                    cv::imwrite(out_filename, mat_color, compression_params);
+            }
             else
-            cv::imwrite(out_filename, mat_output, compression_params);
+            {
+                if(DISPLAY)
+                    cv::imshow("output", mat_output[idx]);
+                else
+                    cv::imwrite(out_filename, mat_output[idx], compression_params);
```

Comment:
We are writing the every batch i.e a single tensor into a single jpeg file.
Since there can be multiple output tensors in the pipeline we have created multiple cv matrix for every outputs.